manifests/generic-pipelines.js

```diff
@@ -194,4 +194,125 @@ async function timedObservationViewer(fileManager, manifest) {
   return { headers, items }
 }
 
-export { genericDateViewer, timedObservationViewer }
+const isValidLat = num => isFinite(num) && Math.abs(num) <= 90
+const isValidLon = num => isFinite(num) && Math.abs(num) <= 180
+const namesLat = ['lat']
+const namesLon = ['lon', 'lng']
+
+function extractJsonLocations(items) {
+  function recurse(node, path) {
+    if (isObject(node)) {
+      // Object
+
+      // First check if there is possible lat and lon at this level
+      let latHeader = null
+      let lonHeader = null
+      Object.entries(node).forEach(([k, v]) => {
+        const kLw = k.toLowerCase()
+        // Specific to Google format, lat and lon are multiplied by 1e7
+        if (kLw.includes('e7')) node[k] = v = v * 1e-7
```

Comment:
I wonder if this mutation propagates to the `#jsonItems` dictionary of the `fileManager` and affects other experiences as well

Reply:
That's something to watch out for indeed, but the only call of this function is on line 304

`extractJsonLocations(JSON.parse(json))`

where json is obviously a string, so there's nothing to worry about